### PR TITLE
gitignore: Ignore all build** trees

### PR DESCRIPTION
**Pull request by tdewey-rpi**
When building with QtCreator, or another CMake-driving target, you may find the IDE creates build directories of the form:

build**

Typically, the IDE appends some kind of config string relevant to the configuration it's building (eg, RelWithDebInfo).

Unfortunately, none of these directories should be commited to the repo.

### CODE REVIEW

**Comment by tdewey-rpi** [comment]
@maxnet If you've no objections, I'm minded to get this in today.

**Comment by maxnet** [comment]
>@maxnet If you've no objections, I'm minded to get this in today.

No objections.

Although, I am personally a fan of checking in the individual files you meant to change, instead of working with gitignore's and "git commit -a"'ing. ;-)

**Comment by tdewey-rpi** [comment]
I completely understand that - but with certain IDEs' concept of change tracking (*cough* vscode *cough*) you get presented a list of all 'new' files - including those not previously tracked.

tldr; vscode users will greatly appreciate this!
